src/test/java/com/aws/greengrass/mqttclient/MqttClientTest.java

```diff
@@ -1010,130 +1011,151 @@ void GIVEN_message_size_exceeds_max_limit_WHEN_publish_THEN_future_complete_exce
+                        "mqtt5"),
+                // mqtt shared subscription topic that's 1 byte greater than 256 bytes
+                Arguments.of("$share/share_name/my/example/topic/thats/too/large/00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
+                        "mqtt3"),
+                Arguments.of("$share/share_name/my/example/topic/thats/too/large/00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
+                        "mqtt5"),
+                // other reserved topic that's 1 byte greater than 512 bytes (arbitrary limit)
+                // rather than having to maintain prefixes for every possibility,
+                // rely on server-side validation
+                Arguments.of("$aws/some/other/reserved/topic/too/large/0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
+                        "mqtt3"),
+                Arguments.of("$aws/iotwireless/events/eventName/eventType/sidewalk/resourceType/resourceId/0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
+                        "mqtt5"),
+                // unreserved topic that's 1 byte greater than 256 bytes
+                Arguments.of("my/example/topic/thats/too/large/00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
+                        "mqtt3"),
+                Arguments.of("my/example/topic/thats/too/large/00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
```

Comment:
nit: parameterize the strings of 0s for 256 and 512 bytes and include them using a String.format

Reply:
Good point will use padding to clean this up